### ISSUE TITLE
generating deployit-manifest.xml at buildtime 

### ISSUE THREAD

**Issue by iaon** [xebialabs-community/gradle-xld-plugin#5]
Hello,

I'm trying to implement following workflow
1. Build the atrifact
2. Generate deployit-manifest.xml
3. Build DAR package
4. Deploy DAR into Sonatype Nexus
5. Deploy DAR file into XL deploy

Because I have dozens of artifacts which have dozens of parameters so I can't hardcode them into static manifest files.

Looks like this way impossible with current version of gradle-xld-plugin
The plugin is trying parse Manifest file during configuration phase before build 

```
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':XXXXX-data-db'.
> DAR manifest file does not exist: [/home/XXXXX/git/tibco-esb/DTL/XXXXX/XXXXX-data-db/build/deployit-manifest.xml]
   > /home/XXXXX/git/tibco-esb/DTL/XXXXX/XXXXX-data-db/build/deployit-manifest.xml

```

Or maybe I'm wrong?
Do you have any suggestion?

Thanks

**Comment by byaminov** [comment]
Hi Ivan,

Yes, indeed manifest file is processed now during gradle project evaluation phase. This is done so that if for example you want to put an output of another task into .dar file:

```
<jee.War name="HelloDeployment" file="${artifact(project.war)}" />
```

then `dar` task automatically starts depending on the corresponding output task: `war` in this example:

```
> ./gradlew dar
:compileJava UP-TO-DATE
:processResources UP-TO-DATE
:classes UP-TO-DATE
:war
:dar
```

If manifest is processed later, e.g. during task execution phase, then it's too late to insert a `war` task into task graph.

If you really need to generate your manifest files during gradle execution phase, then I could make this configurable in the `dar` task via an extension or task parameter. But then you would need to manually add dependencies, like "`dar.dependsOn war`" in my example.

So do you need this `evaluateManifestDuringExecutionPhase = true` parameter to be implemented in `gradle-xld-plugin`, or did you manage to solve your situation in a different manner?

**Comment by iaon** [comment]
Hi Bulat,

Thanks for the explanation. In fact I've solved my problem by Jenkins XLD plugin and some Jenkins jobs, but I don't think it's a good design. I prefer to have all code in the same place.
If you put this task into backlog, it will be very nice.

**Comment by byaminov** [comment]
@iaon ok, I'll have a look when I have time.

**Comment by iaon** [comment]
Thank you!

**Comment by ablaagaard** [comment]
Hi,
I ran into the same issue - I was trying to create a Gradle plugin that, among other stuff, would generate the deployit-manifest.xml, because I have a lot of projects that have identical deployit-manifest.xml except for the project name and version

**Comment by ypersieljp** [comment]
Hello, indeed i thought it was the main core idea of the plugin ...
When dev develops they don t care to modify the manifest and it is updated automatically .. and with extra luck if we put a determined template to follow during the build it remember the developer that it forget to use token or put some text files with necessary information ....

**Comment by byaminov** [comment]
Hi everybody, I'll see what I can do to make the `deployit-manifest.xml` "generateable".

**Comment by ablaagaard** [comment]
Just to clarify - in my case I don't strictly need the deployment manifest to be "generateable", although that would also work, I'm just looking for "shareable"

**Comment by byaminov** [comment]
@ablaagaard sharing of `deployit-manifest.xml` is already possible in current version of the plugin: just configure its location in your `build.gradle` (also documented in [the README](https://github.com/xebialabs-community/gradle-xld-plugin#dar-task)):

```
xldeploy {
    manifest = file('my/shared/deployit-manifest.xml')
}
```

**Comment by byaminov** [comment]
@ypersieljp @iaon @ablaagaard I released a new version of the plugin: [0.3.0](https://plugins.gradle.org/plugin/com.xebialabs.xl-deploy/0.3.0) with the ability to generate `deployit-manifest.xml`. 

You can check an example how it's done in [this test script](https://github.com/xebialabs-community/gradle-xld-plugin/blob/master/src/test/resources/HelloDeployment/generated_deployment.gradle#L40). Note however that if the manifest file is generated, the plugin cannot automatically determine if another task generates artifacts for it. So in my example you have to manually specify that `dar.dependsOn war`.

Please let me know if this works for you!

**Comment by jdewinne** [comment]
Close inactive issue
